lib/include/openamp/virtio.h

```diff
@@ -34,8 +34,19 @@ extern "C" {
 #define VIRTIO_CONFIG_STATUS_FAILED    0x80
 
 /* Virtio device role */
-#define VIRTIO_DEV_MASTER	0UL
-#define VIRTIO_DEV_SLAVE	1UL
+#define VIRTIO_DEV_DRIVER	0UL
+#define VIRTIO_DEV_DEVICE	1UL
+
+#ifdef VIRTIO_MASTER_ONLY
+#define VIRTIO_DRIVER_ONLY
+#warning "VIRTIO_MASTER_ONLY is deprecated, please use VIRTIO_DRIVER_ONLY"
+#endif
+
+#ifdef VIRTIO_SLAVE_ONLY
+#define VIRTIO_DEVICE_ONLY
+#define DO_PRAGMA(x) _Pragma(#x)
```

Comment:
this line seems useless

Reply:
I have taken a look at Zephyr CI fails and would like to suggest an update to your RP.
I have added a fix on top of your patch that solves the `'RPMSG_MASTER' undeclared` [issue](https://github.com/OpenAMP/open-amp/runs/6205417148?check_suite_focus=true#step:5:25772).
Please find my update on my github https://github.com/arnopo/open-amp/pull/17 with associated CI that is successful. 

The error become a warning:
```
/github/workspace/zephyrproject/zephyr/samples/subsys/ipc/openamp/src/main.c:229:2: warning: 'deprecated_rpmsg_master' is deprecated [-Wdeprecated-declarations]
  229 |  vdev.role = RPMSG_MASTER;
      |  ^~~~
In file included from /github/workspace/zephyrproject/modules/lib/open-amp/open-amp/lib/include/openamp/open_amp.h:12,
                 from /github/workspace/zephyrproject/zephyr/samples/subsys/ipc/openamp/src/main.c:18:
/github/workspace/zephyrproject/modules/lib/open-amp/open-amp/lib/include/openamp/rpmsg_virtio.h:84:32: note: declared here
   84 | __deprecated static inline int deprecated_rpmsg_master(void)
      |                                ^~~~~~~~~~~~~~~~~~~~~~~
``` 
This allows backwards compatibility with the deprecated mechanism. and it seems simple enough to understand the problem by looking in the deprecated_rpmsg_master function.

If you are agree with this approach don't hesitate to squash the update in your patch.

Reply:
I like this approach. I have added it, and squashed it into my patch, with your Signed-off.